From 223a581165d0ef29123b2bb47ba9025dee359443 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 16 Nov 2012 13:42:17 -0800
Subject: [PATCH 084/518] MIPS: Octeon: Only enable flash support if CONFIG_MTD
 is set.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 404c2fccc2a65fc2bdb967a8c074eb561b3c791c
Description:

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/cavium-octeon/Makefile | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/mips/cavium-octeon/Makefile b/arch/mips/cavium-octeon/Makefile
index 3cdd7a2..74c0e37 100644
--- a/arch/mips/cavium-octeon/Makefile
+++ b/arch/mips/cavium-octeon/Makefile
@@ -13,16 +13,17 @@ CFLAGS_octeon-platform.o = -I$(src)/../../../scripts/dtc/libfdt
 CFLAGS_setup.o = -I$(src)/../../../scripts/dtc/libfdt
 
 obj-y := cpu.o setup.o serial.o octeon-platform.o octeon-irq.o csrc-octeon.o
-obj-y += dma-octeon.o flash_setup.o
+obj-y += dma-octeon.o
 obj-y += csrc-octeon-ptp.o
 obj-y += octeon-pci-console.o
 obj-y += octeon-memcpy.o
 obj-y += executive/
 
-obj-$(CONFIG_SMP)		      += smp.o
-obj-$(CONFIG_OCTEON_ILM)	      += oct_ilm.o
-obj-$(CONFIG_SYSFS)                   += octeon-power-throttle.o
+obj-$(CONFIG_SMP)			+= smp.o
+obj-$(CONFIG_OCTEON_ILM)		+= oct_ilm.o
+obj-$(CONFIG_SYSFS)			+= octeon-power-throttle.o
 obj-$(CONFIG_CAVIUM_OCTEON_NAND)	+= octeon-nand.o
+obj-$(CONFIG_MTD)			+= flash_setup.o
 
 DTS_FILES = octeon_3xxx.dts octeon_68xx.dts
 DTB_FILES = $(patsubst %.dts, %.dtb, $(DTS_FILES))
-- 
1.9.1

